pallets/automation-time/src/lib.rs

```diff
@@ -411,32 +413,64 @@ pub mod pallet {
 				T::EnsureProxy::ensure_ok(schedule_as_account, who.clone())?;
 			}
 
-			let destination =
-				MultiLocation::try_from(*destination).map_err(|()| Error::<T>::BadVersion)?;
-			let schedule_fee =
-				MultiLocation::try_from(*schedule_fee).map_err(|()| Error::<T>::BadVersion)?;
+			let destination_location = MultiLocation::try_from(*destination.clone())
+				.map_err(|()| Error::<T>::BadVersion)?;
+			let schedule_fee_location = MultiLocation::try_from(*schedule_fee.clone())
+				.map_err(|()| Error::<T>::BadVersion)?;
 
-			let execution_fee: AssetPayment = *execution_fee;
+			let execution_fee_payment: AssetPayment = *execution_fee.clone();
 			let execution_fee_location =
-				MultiLocation::try_from(execution_fee.clone().asset_location)
+				MultiLocation::try_from(execution_fee_payment.clone().asset_location)
 					.map_err(|()| Error::<T>::BadVersion)?;
 
-			Self::ensure_supported_execution_fee_location(&execution_fee_location, &destination)?;
+			Self::ensure_supported_execution_fee_location(
+				&execution_fee_location,
+				&destination_location,
+			)?;
 
 			let action = Action::XCMP {
+				destination: destination_location,
+				schedule_fee: schedule_fee_location,
+				execution_fee: execution_fee_payment,
+				encoded_call: encoded_call.clone(),
+				encoded_call_weight: encoded_call_weight.clone(),
+				overall_weight: overall_weight.clone(),
+				schedule_as: schedule_as.clone(),
+				instruction_sequence: instruction_sequence.clone(),
+			};
+
+			let schedule_value = schedule.clone().validated_into::<T>()?;
+
+			// Schedule the task.
+			let task_id: TaskIdV2 = Self::validate_and_schedule_task(
+				action.clone(),
+				who.clone(),
+				schedule_value,
+				vec![],
+			)?;
+
+			// Tranform the call into a runtime call and encode it.
+			let call: <T as crate::Config>::RuntimeCall = Call::schedule_xcmp_task {
+				schedule,
 				destination,
 				schedule_fee,
 				execution_fee,
 				encoded_call,
 				encoded_call_weight,
 				overall_weight,
-				schedule_as,
 				instruction_sequence,
-			};
+				schedule_as,
+			}
+			.into();
+			let encoded_call = call.encode();
 
-			let schedule = schedule.validated_into::<T>()?;
+			Self::deposit_event(Event::<T>::TaskScheduled {
```

Comment:
Deposit `TaskScheduled` event

Reply:
If we need to throw an event in the `validate_and_schedule_task` function, we should pass the pre-constructed `encoded_call` into the `validate_and_schedule_task` function.

This is because the `encoded_call` for `schedule_xcmp_task` needs to be constructed using the original function parameters and cannot be built within the `validate_and_schedule_task` function by using the `action` parameter passed to it.

That's why I opted to split `Self::deposit_event(Event::<T>::TaskScheduled { ... })` out this function.

Hi @chrisli30 , what are your thoughts on this?

Reply:
Okay!
However, the name "schedule_task" has already been used.